src/index.d.ts

```diff
@@ -53,6 +53,11 @@ declare namespace RamdaAdjunct {
          */
         isNotArray(val: any): boolean;
 
+        /**
+         * Checks if input value is a non empty `Array`.
+         */
+        isNonEmptyArray(val: any): val is Array<any>;
```

Comment:
@BjornMelgaard I'm not sure about this. Can you advice ? Does this signature make sense in the context of empty array ?

Reply:
@char0n I think yes, there is no way we can express that array is non-empty in types

Reply:
Wouldn't be better to use `boolean` instead ?

Reply:
Look
https://www.typescriptlang.org/play/index.html#src=declare%20function%20isNonEmptyArray(val%3A%20any)%3A%20val%20is%20Array%3Cany%3E%3B%0D%0A%0D%0Afunction%20createName(name%3A%20any)%20%7B%0D%0A%20%20%20%20if%20(isNonEmptyArray(name))%20%7B%0D%0A%20%20%20%20%20%20%20%20return%20name.join()%0D%0A%20%20%20%20%7D%0D%0A%20%20%20%20else%20%7B%0D%0A%20%20%20%20%20%20%20%20return%20name.join(%22%20%22)%3B%0D%0A%20%20%20%20%7D%0D%0A%7D

when `isNonEmptyArray` returns `false` array methods still can be applied on `name`

so `val is Array<any>` if fine )